src/connector/src/sink/formatter/mod.rs

```diff
@@ -426,17 +468,22 @@ macro_rules! dispatch_sink_formatter_str_key_impl {
     ($impl:expr, $name:ident, $body:expr) => {
         match $impl {
             SinkFormatterImpl::AppendOnlyJson($name) => $body,
+            SinkFormatterImpl::AppendOnlyBytesJson($name) => $body,
             SinkFormatterImpl::AppendOnlyTextJson($name) => $body,
             SinkFormatterImpl::AppendOnlyAvro(_) => unreachable!(),
             SinkFormatterImpl::AppendOnlyTextAvro($name) => $body,
+            SinkFormatterImpl::AppendOnlyBytesAvro($name) => $body,
             SinkFormatterImpl::AppendOnlyProto($name) => $body,
             SinkFormatterImpl::AppendOnlyTextProto($name) => $body,
+            SinkFormatterImpl::AppendOnlyBytesProto($name) => $body,
 
             SinkFormatterImpl::UpsertJson($name) => $body,
             SinkFormatterImpl::UpsertTextJson($name) => $body,
             SinkFormatterImpl::UpsertAvro(_) => unreachable!(),
             SinkFormatterImpl::UpsertTextAvro($name) => $body,
+            SinkFormatterImpl::UpsertBytesAvro($name) => $body,
             SinkFormatterImpl::UpsertTextProto($name) => $body,
+            SinkFormatterImpl::UpsertBytesProto($name) => $body,
```

Comment:
FYI to fix the trait issue these shall be `unreachable` in `dispatch_sink_formatter_str_key_impl` similar to the `AppendOnlyAvro` branch.